clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4788,8 +4788,7 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
       FwdDeclEmitter.Visit(K.NameType);
   O << "\n";
 
-  O << "__SYCL_INLINE_NAMESPACE(cl) {\n";
-  O << "namespace sycl {\n";
```

Comment:
What is the problem&

Reply:
The problem is that the integration header includes `defines_elementary.hpp` which selects either SYCL 1.2.1 of SYCL2020 style of namespaces before we know which one should be used. So, it seems it was a bad idea to use the same macro in the integration header and for headers. Going to introduce a separate macro for integration header.

@erichkeane @premanandrao Does it sound OK?

Reply:
That sounds like that should be alright... we just have to make sure the two stay in sync.  Though, I'm not sure how you can get a macro here to work if it depends on a decision that isn't made yet.  Should `defines_elementary.hpp` be doing all of the lifting for this set of namespaces?

Reply:
Can we just use `inline namespace cl` w/o any macro defined in DPC++ headers?

Reply:
>> That sounds like that should be alright... we just have to make sure the two stay in sync.
>> Though, I'm not sure how you can get a macro here to work if it depends on a decision that isn't made yet.
>>  Should defines_elementary.hpp be doing all of the lifting for this set of namespaces?

I'm thinking about separating those namespaces. For integration header we could have a macro in the `defines_elementary.hpp` which, for example, defines to `__sycl_int_header_ns` (it's not quite suitable for the integration footer though). For headers we could you an additional header(`defines_sycl_ns.hpp`) which is not included from the integration header - when we include it we know which namespace version we should use.

>> Can we just use inline namespace cl w/o any macro defined in DPC++ headers?

We can, but we cannot use `inline namespace cl`, because user space will be still poluted.  We can use a different namespace - `__sycl_int_header_ns`, but with a macro it would be easier to be in sync without FE modifications.
This sounds much easier though.